packages/cli/src/lib/__tests__/exec.test.js

```diff
@@ -0,0 +1,53 @@
+import { vol } from 'memfs'
+
+import { runScriptFunction } from '../exec'
+
+jest.mock('fs', () => require('memfs').fs)
+
+const redwoodProjectPath = '/redwood-app'
+process.env.RWJS_CWD = redwoodProjectPath
+
+describe('exec.js', () => {
+  beforeEach(() => {
+    vol.reset()
+    jest.clearAllMocks()
+  })
+
+  afterEach(() => {
+    vol.reset()
+    jest.clearAllMocks()
+  })
+
+  afterAll(() => {
+    jest.restoreAllMocks()
+  })
+
+  describe('runScriptFunction', () => {
+    it('resolves and runs script with no extension', async () => {
+      const expectedScriptReturn = Math.floor(Math.random() * 1000)
+      const fooScript = `
+        default async ({ args }) => ${expectedScriptReturn}
+        module.exports = default
+      `
+
+      vol.fromNestedJSON(
+        {
+          'redwood.toml': '',
+          scripts: {
+            'foo.js': fooScript,
+          },
+        },
+        redwoodProjectPath
+      )
+
+      // NOTE: this test currently fails with:
+      // "Error: Cannot find module '/redwood-app/scripts/foo.js' from 'src/lib/exec.js'"
+      await expect(
+        runScriptFunction({
+          path: '/redwood-app/scripts/foo',
+          functionName: 'default',
+        })
+      ).resolves.toBe(expectedScriptReturn)
```

Comment:
I am not sure how to test running the actual script, at least not when using memfs. 

The code in question calls `require('/redwood-app/scripts/foo.js'`, but this fails to load the test script module, probably because the test process is running within the context of the actual framework directory (e.g. '/Users/user/dev/redwoodjs/redwood'), not from the imaginary 'memfs' file system.

Reply:
Hi @Tobbe, in this case should I look to modify/add an integration test?

Reply:
I found https://github.com/streamich/fs-monkey/blob/master/docs/api/patchRequire.md to help with the `require` problem 🙂 Take a look in the tests I pushed